uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Page/Error404Checker.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+require_once 'CRM/Core/Page.php';
+
+/**
+ * Handles CiviCRM paths that are not registered in menu table.
+ */
+class CRM_HRCore_Page_Error404Checker extends CRM_Core_Page {
+  /**
+   * List of URL's that can be shown in 404 error page to suggest to user, of the
+   * form 'Link Label' => 'URL'
+   * 
+   * @var array
+   */
+  private $_suggestedUrls = [];
+
+  /**
+   * Constructor, populates $_suggestedUrls array.
+   * 
+   * @param string $title
+   *   Title of the page.
+   * @param int $mode
+   *   Mode of the page.
+   */
+  public function __construct($title = NULL, $mode = NULL) {
+    $this->_suggestedUrls = [
+      'CiviHR Dashboard' => CRM_Utils_System::url('civicrm/tasksassignments/dashboard', 'reset=1'),
+      'Search Contacts' => CRM_Utils_System::url('civicrm/contact/search', 'reset=1'),
+      'Vacancies Dashboard' => CRM_Utils_System::url('civicrm/vacancy/dashboard', 'reset=1'),
+    ];
+
+    parent::__construct($title, $mode);
+  }
+  
+  /**
+   * Checks if path is either 'civicrm', to redirect to T&A Dashboard, or 
+   * 'civicrm/dashboard', to redirect to CiviCRM Contact Dashboard.  Shows 404
+   * error page otherwise.
+   */
+  function run() {
+    $path = implode('/', $this->urlPath);
+
+    switch ($path) {
+      case 'civicrm':
+        CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/tasksassignments/dashboard', 'reset=1'));
+        break;
+      case 'civicrm/dashboard':
+        CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/default/dashboard', 'reset=1'));
+        break;
+      default:
+        CRM_Utils_System::setTitle(ts('404 - Page not Found!'));
+        self::$_template->assign('urls', $this->_suggestedUrls);

```

Comment:
$_template is declared as static in perent class:
https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Page.php#L87